network/network.go

```diff
@@ -1075,3 +1131,17 @@ func (n *network) disconnected(p *peer) {
 		n.router.Disconnected(p.id)
 	}
 }
+
+// assumes stateLock is held
+// iterate the peers and find if we are already peered to this node.
+func (n *network) IsPeered(id ids.ShortID) bool {
```

Comment:
Could we un-export this? If it assumes the stateLock is held, it should probably only be called internally.